cloudinit/net/network_state.py

```diff
@@ -989,7 +989,7 @@ def _normalize_net_keys(network, address_keys=()):
 
     @returns: A dict containing normalized prefix and matching addr_key.
     """
-    net = dict((k, v) for k, v in network.items() if v)
+    net = {k: v for k, v in network.items() if v or v == 0}
```

Comment:
This would capture other falsey values as `"0"`.

```suggestion
    net = {k: v for k, v in network.items() if v is not None}
```

Reply:
"0" isn't truthy in Python. I think we do want to capture other falsey values here. 0 is the only one we want to pass through.

Reply:
My bad here.